### PR TITLE
build: use npm gradle plugin

### PR DESCRIPTION
**Pull request by kelvin-olaiya**
When reviewing, please look if the dependencies are ok for the npm package.
I've copied those from the various package.json and refactor them.

### CODE REVIEW

**Comment by kelvin-olaiya** [comment]
Ideally, with this PR, we could stop tracking the `package.json` and `package-lock.json` files (so no worries if you notice something strange in those files now). Before doing that, I must make sure that renovate will be able to identify dependencies with this new system.

**Comment by kelvin-olaiya** [comment]
I've updated the renovate configuration, so now the various `package.Json` files can be untracked

**Comment by kelvin-olaiya** [comment]
As you can see [here](https://github.com/kelvindev15/test-renovate-custom-manager/pull/2/files), renovate will be able to update versions.

File: alarm/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+packageJson {
+    name = "alarm"
+    description = "Alarm microservice"
+    main = "dist/src/index.js"
+    scripts {
+        val scriptDeps = listOf(npmScript("build") inProject "common")
+        listOf(
+            "test" runs "vitest --exclude \"test/tolerance/**\" --run",
```
Comment (paga16-hash) [inline alarm/build.gradle.kts:12]: i think we can remove the --exclude ...

File: auth/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+plugins {
+    id("configure-npm")
+}
+
+packageJson {
```
Comment (paga16-hash) [inline auth/build.gradle.kts:5]: oh mah gah

**Review by paga16-hash — DISMISSED** [review]

**Review by Mala1180 — DISMISSED** [review]

File: alarm/build.gradle.kts (hunk 1 above)
Comment (kelvin-olaiya) [reply alarm/build.gradle.kts]: @paga16-hash it seems that the tolerance tests are not passing...

**Review by Mala1180 — APPROVED** [review]

**Pull request merged**